backend/service/xkcd.go

```diff
@@ -0,0 +1,88 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+	"xkcd/comics/model"
+)
+
+const xkcdURL = "https://xkcd.com/%d/info.0.json"
+
+func GetComic(comicNum int) (*model.Comic, error) {
+    resp, err := http.Get(fmt.Sprintf(xkcdURL, comicNum))
+    if err != nil {
+        return nil, err
+    }
+    defer resp.Body.Close()
+
+    var comic model.Comic
+    if err := json.NewDecoder(resp.Body).Decode(&comic); err != nil {
+        return nil, err
+    }
+
+    return &comic, nil
```

Comment:
There's no need to return a pointer to the struct here. If possible we want to not do this unless necessary, as it would end up escaping to the heap and perform unnecessary memory allocations.

Have a look [here](https://landontclipp.github.io/blog/2023/07/15/analyzing-go-heap-escapes/) for more info.